src/vs/workbench/contrib/terminal/browser/xterm/xtermTerminal.ts

```diff
@@ -169,9 +171,7 @@ export class XtermTerminal extends DisposableStore implements IXtermTerminal {
 		this._updateUnicodeVersion();
 		this._commandNavigationAddon = this._instantiationService.createInstance(CommandNavigationAddon, _capabilities);
 		this.raw.loadAddon(this._commandNavigationAddon);
-		this._shellIntegrationAddon = this._instantiationService.createInstance(ShellIntegrationAddon);
-		this.raw.loadAddon(this._shellIntegrationAddon);
-		this._updateDecorationAddon();
```

Comment:
We always want to be listening to the shell integration sequences such that the terminal will still activate shell integration capabilities when it is installed manually into shell scripts. I think we should move the telemetry part into `ShellIntegrationAddon` because of this, and also include an event for when shell integration gets activated when the args were not injected.

It would also be good to have it in there since it's a shell integration-specific event.

Reply:
Could we pass in telemetry service directly then as `ITelemetryService | undefined` and use undefined in the pty host?

Reply:
pretty sure I tried that and it didn't work for some reason, but will try again